src/dispatch/batch_builder.rs

```diff
@@ -0,0 +1,150 @@
+use crate::{
+    dispatch::{Dispatcher, DispatcherBuilder},
+    world::ResourceId,
+    Accessor, AccessorCow, DynamicSystemData, RunningTime, System, SystemData, World,
+};
+
+/// The `BatchBuilder` is responsible for the creation of the Batch.
+///
+/// The `Batch` is a Systam that have the capabilities to contain another dispatcher
+/// Thanks to this characteristic is possible to customize its dispatching
+///
+/// The BatchBuilder accept the dispatcher builder as parameter, and the type of the
+/// System that will drive the execution of the internal dispatcher.
+///
+/// Note that depending on the dependencies of the SubSystems the Batch
+/// can run in parallel with other Systems.
+/// In addition the Sub Systems can run in parallel within the Batch.
+#[derive(Default)]
+pub struct BatchBuilder<'a, 'b> {
+    dispatcher_builder: DispatcherBuilder<'a, 'b>,
+}
+
+impl<'a, 'b> BatchBuilder<'a, 'b> {
```

Comment:
Hmm, on a second thought, we could probably just add a method `build_batch` to the `DispatcherBuilder`, then we don't even need this extra type.

Reply:
Ok I'll add it